backend/routes/gallery/tokens.js

```diff
@@ -0,0 +1,37 @@
+  import express from "express";
+  import jwt from "jsonwebtoken";
+  import dotenv from "dotenv";
+  dotenv.config({ path: '../../.env' });
+
+  const router = express.Router();
+
+  // Generate token based on user IP
+  router.get("/", (req, res) => {
+    try {
+      const userIP = req.ip;
+      // Create a token with user IP
+      const token = jwt.sign({ ip: userIP }, process.env.SECRET_KEY, { expiresIn: "1h" });
+      console.log("ip: "+ userIP);
+      console.log("access token: " + token);
+
+      // Set the token as a cookie
+      try{
+        res.cookie("access_token", token, {
+          httpOnly: true,
+          secure: false,
+          maxAge: 360000,  // 1 hr
```

Comment:
we can probably do longer max age than this for our use case

Reply:
ok, I'll update it for like 1 day I guess.